Linux:  Could not install Couchbase Lite native libraries: Cannot create link to target 'libcblite.so.3'

**Describe the bug**

With the following pubspec.yaml the error occures:
```
  cbl: ^3.4.0
  cbl_flutter: ^3.3.0
  cbl_flutter_ce: ^3.3.1
```
The error during the build is this:

```
CMake Error at flutter/ephemeral/.plugin_symlinks/cbl_flutter_ce/linux/CMakeLists.txt:41 (message):
  Could not install Couchbase Lite native libraries:

  255



  Unhandled exception:

  PathExistsException: Cannot create link to target 'libcblite.so.3', path =
  '/home/machine/.pub-cache/hosted/pub.dev/cbl_flutter_ce-3.3.1/linux/lib/libcblite.so'
  (OS Error: File exists, errno = 17)

  #0 _checkForErrorResponse (dart:io/common.dart:58:9)

  #1 _Link.create.<anonymous closure> (dart:io/link.dart:261:11)

  <asynchronous suspension>

  #2 copyDirectoryContents (package:cbl/src/install/utils.dart:159:9)

  <asynchronous suspension>

  #3 main
  (file:///home/machine/.pub-cache/hosted/pub.dev/cbl_flutter_install-0.1.0+1/bin/cbl_flutter_install.dart:65:7)


  <asynchronous suspension>
```

**To Reproduce**

Steps to reproduce the behavior:
1. add the cbl packages to a project with linux as target (dev machine is also linux)
2. run `flutter build linux`

**Expected behavior**

The cbl native libs are installed without an error. 
However, the installation can be fixed by temporary deleting the libs:
`rm -rf ~/.pub-cache/hosted/pub.dev/cbl_flutter_ce-3.3.1/linux/lib/libcblite* `

**Screenshots**

If applicable, add screenshots to help explain your problem.

**Environment (please complete the following information):**

 - OS: linux (ubuntu/arch)
 - `cbl`: 3.4.0
 - `cbl_flutter`: 3.3.0
 - `cbl_flutter_ce`: 3.3.1

--- a/packages/cbl/lib/src/install/utils.dart
+++ b/packages/cbl/lib/src/install/utils.dart
@@ -156,7 +156,13 @@ Future<void> copyDirectoryContents(
         await File(destPath).create(recursive: true);
         await File(entity.resolveSymbolicLinksSync()).copy(destPath);
       } else {
-        await Link(destPath).create(await entity.target());
+        try {
+          await Link(destPath).create(await entity.target());
+        } on PathExistsException {
+          // Links can't be overwritten on some platforms.
+          await Link(destPath).delete();
+          await Link(destPath).create(await entity.target());
+        }
       }
     } else if (entity is File) {
       await entity.copy(destPath);

--- a/packages/cbl/test/install/package_test.dart
+++ b/packages/cbl/test/install/package_test.dart
@@ -3,12 +3,12 @@ import 'dart:io';
 import 'package:cbl/src/install/package.dart';
 import 'package:test/test.dart';
 
+import '../utils.dart';
+
 void main() {
   group('DatabasePackageConfig', () {
     test('download packages ', () async {
-      final tempCacheDir = Directory.systemTemp.createTempSync();
-      addTearDown(() => tempCacheDir.deleteSync(recursive: true));
-
+      final tempCacheDir = tempTestDirectory();
       final loader = RemotePackageLoader(cacheDir: tempCacheDir.path);
       final configs = [
         for (final edition in Edition.values)
@@ -23,9 +23,7 @@ void main() {
 
   group('VectorSearchPackageConfig', () {
     test('download packages', () async {
-      final tempCacheDir = Directory.systemTemp.createTempSync();
-      addTearDown(() => tempCacheDir.deleteSync(recursive: true));
-
+      final tempCacheDir = tempTestDirectory();
       final loader = RemotePackageLoader(cacheDir: tempCacheDir.path);
       final packageConfigs = VectorSearchPackageConfig.all(release: '1.0.0')
           .withoutMacOSonDifferentHost;

--- a/packages/cbl/test/install/utils_test.dart
+++ b/packages/cbl/test/install/utils_test.dart
@@ -1,0 +1,26 @@
+import 'dart:io';
+
+import 'package:cbl/src/install.dart';
+import 'package:test/test.dart';
+
+import '../utils.dart';
+
+void main() {
+  group('copyDirectoryContents', () {
+    test('override link', () async {
+      final sourceDir = tempTestDirectory();
+      final targetDir = tempTestDirectory();
+
+      File('${sourceDir.path}/file').writeAsStringSync('content');
+      Link('${sourceDir.path}/link').createSync('file');
+
+      await copyDirectoryContents(sourceDir.path, targetDir.path);
+      expect(File('${targetDir.path}/file').readAsStringSync(), 'content');
+      expect(Link('${targetDir.path}/link').targetSync(), 'file');
+
+      await copyDirectoryContents(sourceDir.path, targetDir.path);
+      expect(File('${targetDir.path}/file').readAsStringSync(), 'content');
+      expect(Link('${targetDir.path}/link').targetSync(), 'file');
+    });
+  });
+}

--- a/packages/cbl/test/utils.dart
+++ b/packages/cbl/test/utils.dart
@@ -1,0 +1,9 @@
+import 'dart:io';
+
+import 'package:test/test.dart';
+
+Directory tempTestDirectory() {
+  final directory = Directory.systemTemp.createTempSync();
+  addTearDown(() => directory.deleteSync(recursive: true));
+  return directory;
+}

